nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -232,15 +291,17 @@ private void compareNullabilityAnnotations(
         }
       }
       if (isLHSNullableAnnotated != isRHSNullableAnnotated) {
-        reportInvalidAssignmentInstantiationError(tree, lhsType, rhsType, state, analysis);
-        return;
+        return false;
       }
       // nested generics
       if (lhsTypeArgument.getTypeArguments().length() > 0) {
-        compareNullabilityAnnotations(
-            (Type.ClassType) lhsTypeArgument, (Type.ClassType) rhsTypeArgument, tree);
+        isValid =
+            isValid
+                && compareNullabilityAnnotations(
+                    (Type.ClassType) lhsTypeArgument, (Type.ClassType) rhsTypeArgument);
```

Comment:
As an optimization, you should immediately return false if the recursive call to `compareNullabilityAnnotations` returns false.  Like, `if (!compareNullabilityAnnotations(...)) return false;`.  Then we don't need the `isValid` variable